src/CMakeLists.txt

```diff
@@ -587,15 +585,22 @@ if(${WITH_RADOSGW})
     cls_statelog_client
     cls_rgw_client
     fcgi
-    ${CURL_LIBRARIES}
     ${EXPAT_LIBRARIES}
   )
 endif(${WITH_RADOSGW})
-if(${WITH_RBD})
+if(WITH_RBD)
   set(DENCODER_EXTRALIBS
       ${DENCODER_EXTRALIBS}
       rbd_replay_types)
-endif(${WITH_RBD})
+  if(LINUX)
+    list(APPEND dencoder_srcs
+        $<TARGET_OBJECTS:krbd_objs>)
+  endif(LINUX)
+endif(WITH_RBD)
+if(WITH_KVS)
+  list(APPEND dencoder_srcs
+      $<TARGET_OBJECTS:parse_secret_objs>)
```

Comment:
could you explain the rationales in the commit message? imo, it's not self-explained.